### ISSUE TITLE
Login into Home Assistant breaks at a forward

### ISSUE THREAD

**Issue by RubenKelevra** [LadybirdBrowser/ladybird#939]
[Home Assistant](https://github.com/home-assistant/core) is an open-source home automation project with a local first approach. I've been using it for years.

I run it on a Raspberry Pi 4B and connect either through VPN or directly over my local network.

Recently, I tried to log in, and while the initial login worked, it asked for a second factor authentication. However, after entering the second factor, the redirect didn’t work.

I found this curious, as I think you would have expected the redirect to function properly by now.

![Screenshot_20240803_073312](https://github.com/user-attachments/assets/a53c74ab-3131-420f-a16d-e8b5b9c79e28)

**Comment by AtkinsSJ** [comment]
Are you able to isolate what the redirect is?

Also, does this still happen if you run Ladybird with `--enable-qt-networking`?

**Comment by RubenKelevra** [comment]
> Are you able to isolate what the redirect is?

How do I do this?

Or should I run Wireshark to record the conversation?

> Also, does this still happen if you run Ladybird with `--enable-qt-networking`?

Nope. Works fine with that flag:

![Screenshot_20240803_110000](https://github.com/user-attachments/assets/6ef5cd04-bde0-46f8-9ef1-1d96944e1d83)

And after a while the dashboard appears:

![Screenshot_20240803_110155](https://github.com/user-attachments/assets/30502554-c851-4544-a576-03ea5eb7a542)

Btw: Should we report `Unhandled JavaScript exception (in promise)` or are those expected at the moment?

**Comment by RubenKelevra** [comment]
> Or should I run Wireshark to record the conversation?

Here's the login conversation. `REMOVED` means I removed stuff, to get the size down etc. :)

Hope that helps:

[conversation_broken_refer_filtered.txt](https://github.com/user-attachments/files/16478929/conversation_broken_refer_filtered.txt)

**Comment by AtkinsSJ** [comment]
Great! That should be really helpful.

It working fine with Qt networking means it's an issue with our own networking code, so that narrows it down.

**Comment by RubenKelevra** [comment]
I've tried it with Ladybird 34261e54901 and it's still broken:

![Screenshot_20241013_165819](https://github.com/user-attachments/assets/f5f31442-2381-410f-93e9-7ddebbf9fa74)

**Comment by AtkinsSJ** [comment]
> While the workaround with `--enable-qt-networking` does not work anymore:
> 
> ```
> Unrecognized option --enable-qt-networking
> ``` 

Yeah, we replaced the networking stacks with Curl, so that flag doesn't exist any more. Odd that Curl doesn't work here though.

**Comment by RubenKelevra** [comment]
> Yeah, we replaced the networking stacks with Curl, so that flag doesn't exist any more.

I've heard about that, that's why I wanted to recheck.

> Odd that Curl doesn't work here though.

I wonder what's so special about the Home Assistant login...

**Comment by InvalidUsernameException** [comment]
@RubenKelevra Can you retest this with the latest version from `master`? Your last screenshot contains an IPv6 address and there were some fixes related to that (see #3596).
